sway/server.c

```diff
@@ -126,7 +126,7 @@ bool server_init(struct sway_server *server) {
 		&server->layer_shell_surface);
 	server->layer_shell_surface.notify = handle_layer_shell_surface;
 
-	server->xdg_shell = wlr_xdg_shell_create(server->wl_display);
+	server->xdg_shell = wlr_xdg_shell_create(server->wl_display, 2);
```

Comment:
For confirmation, this fixes it. I created a patch in my local Gentoo overlay, and sway compiles and runs successfully with the latest wlroots git commit.

Reply:
> I think it would be more readable to have this as a definition somewhere rather than a magic number. Maybe something like `SWAY_XDG_SHELL_VERSION`.

Sure, done.

> Or maybe even have wlroots define something to indicate the last version available.

Nack, this was specifically implemented to ensure that compositors request a version that they support. Relying on wlroots to provide the latest version while not implementing the necessary compositor changes is a bad idea.